### ISSUE TITLE
compile step fails for sunburst.src.js

### ISSUE THREAD

**Issue by RainerSchwarze** [highcharts/highcharts#7759]
I copied compiler.jar from `./tools/google-closure-compiler/` to the project dir to make things easier.
Then build the project and execute this:

`java -jar compiler.jar --js code\js\modules\sunburst.src.js --js_output_file tmp.js --language_in ES5
`

The output is:

```
code\js\modules\sunburst.src.js:2502: ERROR - Parse error. missing : after property id
                calculateLevelSizes,
                                    ^
code\js\modules\sunburst.src.js:2503: ERROR - Parse error. syntax error
                range: range
                       ^
code\js\modules\sunburst.src.js:2505: ERROR - Parse error. missing ) in parenthetical
        };
          ^
code\js\modules\sunburst.src.js:2584: ERROR - Parse error. missing ) in parenthetical
    }(Highcharts, draw, result));
                                 ^
code\js\modules\sunburst.src.js:2585: ERROR - Parse error. syntax error
5 error(s), 0 warning(s)
```

The problem seems to be this snippet in sunburst.src.js:

```
	utils: {
		calculateLevelSizes,
		range: range
	}
```

#### Product version

latest Highcharts v6

**Comment by pawelfus** [comment]
Hi @RainerSchwarze - and thanks! Could you change `language_in` to ES6? 

*Internal note:*
@jon-a-nygaard shouldn't we use `calculateLevelSizes: calculateLevelSizes`? Or our builder should return ES5?

**Comment by RainerSchwarze** [comment]
I downloaded the latest compiler and got this (the one from the tools dir does not support ES6):

`java -jar cc.jar --js code\js\modules\sunburst.src.js --js_output_file tmp.js --language_in ES6`

```
code/js/modules/sunburst.src.js:752: WARNING - Misplaced type annotation. Type annotations are not allowed here. Are you missing parentheses?
        }, {                                                                                                  
           ^                                                                                                  
code/js/modules/sunburst.src.js:1687: WARNING - Misplaced type annotation. Type annotations are not allowed here. Are you missing parentheses?
    }(Highcharts, result));                                                  
      ^^^^^^^^^^
code/js/modules/sunburst.src.js:2576: WARNING - Misplaced type annotation. Type annotations are not allowed here. Are you missing parentheses?
        seriesType(
        ^^^^^^^^^^^
0 error(s), 3 warning(s)
```

**Comment by RainerSchwarze** [comment]
(I thought that I saw no error output when I ran `gulp compile` the last time, but now I see it - I'm adding it for completeness:)
```
...
Compile error in file: js/modules/sunburst.src.js                                                                                     
- Type: ES6_FEATURE                                                                                                                   
- Line: 2501                                                                                                                          
- Char : 19                                                                                                                           
- Description: this language feature is only supported for ECMASCRIPT6 mode or better: extended object literal. Use --language_in=ECMA
SCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features.                                                                                 
[13:13:29] Finished 'compile' after 6.55 min                                                                                          
```

**Comment by pawelfus** [comment]
Did you try setting `--language_in=ECMASCRIPT6` ? I didn't find anything in the docs about `--language_in=ES6` or `ES5`. Maybe those are shortcuts and I missed them?

**Comment by RainerSchwarze** [comment]
I was looking into custom builds and in the `gulpfile.js` is this snippet:

```
const out = closureCompiler.compile({
    compilationLevel: 'SIMPLE_OPTIMIZATIONS',
    jsCode: [{
        src: src
    }],
    languageIn: 'ES5',
    languageOut: 'ES5',
    createSourceMap: createSourceMap
});
```

Indeed ES5 and ES6 are not shown in the command line help. But they seem to be official - I found mentions in the FAQ: https://github.com/google/closure-compiler/wiki/FAQ

**Comment by jon-a-nygaard** [comment]
Hi @RainerSchwarze.

I will close this issue as it is fixed by commit a35c95f.

I briefly tested `languageIn: 'ES6'` in `gulp compile` and that seemed to work fine. So I am not sure why `java -jar cc.jar --js code\js\modules\sunburst.src.js --js_output_file tmp.js --language_in ES6` caused errors.
Please let us know if there should be further problems with this.

Thanks again for reporting!

**Comment by TorsteinHonsi** [comment]
Fixed by a35c95f07ba16abbef28f9c7d1b8de9a83f9eacf
